src/Controller/EnMarche/CitizenProjectController.php

```diff
@@ -199,4 +204,53 @@ public function listActorsAction(CitizenProject $citizenProject, CitizenProjectM
             'actors' => $citizenProjectManager->getCitizenProjectMemberships($citizenProject),
         ]);
     }
+
+    /**
+     * @Route("/{slug}/acteurs/contact", name="app_citizen_project_contact_actors")
+     * @Method("POST")
+     * @Security("is_granted('ADMINISTRATE_CITIZEN_PROJECT', citizenProject)")
+     */
+    public function contactActorsAction(Request $request, CitizenProject $citizenProject, CitizenProjectManager $citizenProjectManager): Response
+    {
+        $this->disableInProduction();
+
+        if (!$this->isCsrfTokenValid('citizen_project.contact_actors', $request->request->get('token'))) {
+            throw $this->createAccessDeniedException('Invalid CSRF protection token to contact actors.');
+        }
+
+        $uuids = CitizenProjectUtils::getUuidsFromJson($request->request->get('contacts', ''));
+        $adherents = CitizenProjectUtils::removeUnknownAdherents($uuids, $citizenProjectManager->getCitizenProjectMembers($citizenProject));
+        $command = new CitizenProjectContactActorsCommand($adherents, $this->getUser());
+        $contacts = CitizenProjectUtils::getUuidsFromAdherents($adherents);
+
+        if (empty($contacts)) {
+            $this->addFlash('info', $this->get('translator')->trans('citizen_project.contact_actors.none'));
+
+            return $this->redirectToRoute('app_citizen_project_list_actors', [
+                'slug' => $citizenProject->getSlug(),
+            ]);
+        }
+
+        $form = $this->createForm(CitizenProjectContactActorsType::class, $command)
+            ->add('submit', SubmitType::class)
+            ->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get(CitizenProjectContactActorsCommandHandler::class)->handle($command);
+            $this->addFlash('info', $this->get('translator')->trans('citizen_project.contact_actors.success'));
+
+            return $this->redirectToRoute('app_citizen_project_list_actors', [
+                'slug' => $citizenProject->getSlug(),
+            ]);
+        }
+
+        return $this->render('citizen_project/contact.html.twig', [
```

Comment:
This form should be handled in `app_citizen_project_list_actors`.
The logic is duplicated between this two routes and it duplicates content from a SEO point of view.

Reply:
Already seen in PM with @B-Galati.